lib/ethereum.js

```diff
@@ -74,11 +74,36 @@ class Ethereum {
       node = name;
 
     const resolver = await this.getEnsResolver(this.trimDot(node));
+    return await this.resolveDnsFromEnsWithResolver(name, type, node, resolver);
+  }
+
+  async resolveDnsFromEnsWithResolver(name, type, node, resolver) {
     const record = await resolver.dnsRecord(
       this.namehash(this.trimDot(node)),
       this.hashDnsName(name),
       type
     );
+
+    if (!record || record === '0x') {
+      if(type === wire.types.CNAME || type === wire.types.NS) {
+        return null;
+      }
+
+      // CNAME fallback:
+      // Technically, CNAME has priority over
+      // other record types in the zone
+      // but it adds a cost of an additional lookup
+      // to check for CNAME first due to
+      // how EIP-1185 works. For zones setup
+      // correctly this isn't an issue.
+      return await this.resolveDnsFromEnsWithResolver(
```

Comment:
I think you can lose the `await` here, the original caller is already "waiting" for a Promise to resolve.

Reply:
Fixed